src/test/architecture/test_graph_search.py

```diff
@@ -591,3 +688,296 @@ def test_single_node_cyclic_case(self, mocker):
+        mocker.patch.object(comm_mock_1, 'to_value', return_value=comm_mock_struct_1)
+
+        mocker.patch.object(comm_mock_2, 'publish_node_name', '3')
+        mocker.patch.object(comm_mock_2, 'subscribe_node_name', '4')
+        mocker.patch.object(comm_mock_2, 'topic_name', '0->1')
+        mocker.patch.object(comm_mock_2, 'subscription_construction_order', 1)
+        mocker.patch.object(comm_mock_2, 'publisher_construction_order', 1)
+        comm_mock_struct_2 = mocker.Mock(spec=CommunicationStructValue)
+        mocker.patch.object(comm_mock_2, 'to_value', return_value=comm_mock_struct_2)
+
+        mocker.patch.object(comm_mock_3, 'publish_node_name', '4')
+        mocker.patch.object(comm_mock_3, 'subscribe_node_name', '5')
+        mocker.patch.object(comm_mock_3, 'topic_name', '1->2')
+        mocker.patch.object(comm_mock_3, 'subscription_construction_order', 1)
+        mocker.patch.object(comm_mock_3, 'publisher_construction_order', 1)
+        comm_mock_struct_3 = mocker.Mock(spec=CommunicationStructValue)
+        mocker.patch.object(comm_mock_3, 'to_value', return_value=comm_mock_struct_3)
+
+        mocker.patch.object(graph_node_mock_0, 'node_name', node_name_3)
+        mocker.patch.object(graph_node_mock_1, 'node_name', node_name_4)
+        mocker.patch.object(graph_node_mock_2, 'node_name', node_name_5)
+        mocker.patch.object(edge_mock_0, 'node_name_from', node_name_3)
+        mocker.patch.object(edge_mock_0, 'node_name_to', node_name_4)
+        mocker.patch.object(edge_mock_0, 'label', '0->1'+sub_const_name_1+pub_const_name_1)
+        mocker.patch.object(edge_mock_0, 'node_from', graph_node_mock_0)
+        mocker.patch.object(edge_mock_0, 'node_to', graph_node_mock_1)
+        mocker.patch.object(edge_mock_1, 'node_name_from', node_name_4)
+        mocker.patch.object(edge_mock_1, 'node_name_to', node_name_5)
+        mocker.patch.object(edge_mock_1, 'label', '1->2'+sub_const_name_1+pub_const_name_1)
+        mocker.patch.object(edge_mock_1, 'node_from', graph_node_mock_1)
+        mocker.patch.object(edge_mock_1, 'node_to', graph_node_mock_2)
+        mocker.patch.object(
+            graph_path_mock, 'edges', [edge_mock_0, edge_mock_1]
+        )
+        mocker.patch.object(
+            NodePathSearcher, '_create_head_dummy_node_path', return_value=node_path_mock_3)
+        mocker.patch.object(
+            NodePathSearcher, '_create_tail_dummy_node_path', return_value=node_path_mock_5)
+
+        searcher = NodePathSearcher(
+            (node_mock_0, node_mock_1, node_mock_2, node_mock_3, node_mock_4, node_mock_5),
+            (comm_mock_0, comm_mock_1, comm_mock_2, comm_mock_3))
+        path = searcher._to_path(graph_path_mock)
+
+        expected = PathStruct(
+            None, (node_path_mock_3, comm_mock_2, node_path_mock_4, comm_mock_3, node_path_mock_5)
+        )
+        assert path.to_value() == expected.to_value()
+
+    def test_get_sub_pub(self, mocker):
```

Comment:
It is better to move this test to new file dedicated to testing `NodeStruct`.

Reply:
Moved to a new file dedicated to testing NodeStruct